frontend/apps/cloudstorage/cloudstorage.lua

```diff
@@ -0,0 +1,331 @@
+                        })
+                    end
+                    -- FTP
+                    if self.type == "ftp" then
+                        local callback_close = function()
+                            self:onClose()
+                        end
+                        UIManager:scheduleIn(1, function()
+                            Ftp:downloadFtpFile(item, self.address, self.username, self.password, callback_close)
+                        end)
+                        UIManager:close(self.download_dialog)
+                        UIManager:show(InfoMessage:new{
+                            text = _("Downloading may take several minutes..."),
+                            timeout = 1,
+                        })
+                    end
+                end,
+            },
+        },
+        {
+            {
+                text = _("Set download directory"),
+                callback = function()
+                    require("ui/downloadmgr"):new{
+                        title = _("Choose download directory"),
+                        onConfirm = function(path)
+                            self.cs_settings:saveSetting("download_file", path)
+                            self.cs_settings:flush()
+                        end,
+                    }:chooseDir()
+                end,
+            },
+        },
+    }
+    self.download_dialog = ButtonDialog:new{
+        buttons = buttons
+    }
+    UIManager:show(self.download_dialog)
+end
+
+function CloudStorage:onMenuHold(item)
+    if item.editable then
+        local cs_server_dialog
+        cs_server_dialog = ButtonDialog:new{
+            buttons = {
+                {
+                    {
+                        text = _("Info"),
+                        enabled = true,
+                        callback = function()
+                            UIManager:close(cs_server_dialog)
+                            self:infoServer(item)
+                        end
+                    },
+                    {
+                        text = _("Edit"),
+                        enabled = true,
+                        callback = function()
+                            UIManager:close(cs_server_dialog)
+                            self:editCloudServer(item)
+
+                        end
+                    },
+                    {
+                        text = _("Delete"),
+                        enabled = true,
+                        callback = function()
+                            UIManager:close(cs_server_dialog)
+                            self:deleteCloudServer(item)
+                        end
+                    },
+                },
+            }
+        }
+        UIManager:show(cs_server_dialog)
+        return true
+    end
+end
+
+function CloudStorage:editCloudServer(item)
+    item.callback_refresh = self.callback_refresh
+    if item.type == "dropbox" then
+        DropBox:configDropbox(item, self.callback_refresh)
```

Comment:
I feel like having all the provider modules read and write the global cloudserver config is a little bit hacky. Can we keep all the global config management in one place (CloudStorage) instead? For example, you can pass in a editing callback here which does roughly the following:

```lua
function callback(updated_config)
    local servers = self.cs_settings:readSetting("cs_servers") or {}
    for _, server in ipairs(cs_servers) do
        if server.name == updated_config.text and server.password == updated_config.password then
            server.name = fields[1]
            server.password = fields[2]
            break
        end
    end
    self.cs_settings:saveSetting("cs_servers", servers)
    self.callback_refresh()
end
```

Then in provider's config method, you can just call the callback with the updated config. You should be able to do the same for create and delete.